resource/template/resource.go

```diff
@@ -59,6 +65,82 @@ type TemplateResource struct {
 
 var ErrEmptySrc = errors.New("empty src template")
 
+// selectsMe takes a selector and returns
+// true if it selects this node.
+func selectsMe(s string) bool {
+	node := os.Getenv("NODENAME")
+	log.Infof("Checking if %s selects node %s", s, node)
+	ps, err := selector.Parse(s)
+	if err != nil {
+		log.Errorf("Failed to parse selector: %v", err)
+		return false
+	}
+	config, err := apiconfig.LoadClientConfig("")
+	if err != nil {
+		log.Errorf("Failed to load Calico client configuration: %v", err)
+		return false
+	}
+
+	cc, err := clientv3.New(*config)
+	if err != nil {
+		log.Errorf("Failed to create main Calico client: %v", err)
+		return false
+	}
```

Comment:
Wonder if we might want to cache up to here, so as not to repeat for multiple selectsme calls.

Reply:
Yeah, for sure. confd is already caching this data elsewhere I believe, I just didn't spend the time yet to figure out how to join this bit up with the other :)